lib/portlayer/storage/vsphere/image.go

```diff
@@ -732,3 +736,47 @@ func createBaseStructure(op trace.Operation, vmdisk *disk.VirtualDisk) (err erro
 
 	return nil
 }
+
+func (v *ImageStore) StatPath(op trace.Operation, deviceId string, target string) (stat *compute.FileStat, err error){
+
+	host, err := sys.UUID()
+	if err != nil {
+		op.Debugf("Failed to determine host UUID")
+		return nil, err
+	}
+
+	if err = v.verifyImage(op, host, deviceId); err != nil {
+		op.Debugf("Device is not an image")
+		return nil, err
+	}
+
+	// TODO: not sure if these checks are needed
+	//store, err := v.GetImageStore(op, host)
+	//if err != nil {
+	//	op.Debugf("Failed to get image store")
+	//	return nil, err
+	//}
+	//
+	//_, err = v.GetImage(op, store, deviceId)
+	//if err != nil {
+	//	op.Debugf("Device is not an image")
+	//	return nil, err
+	//}
+
+	diskDsURI := v.imageDiskDSPath(host, deviceId)
+	mountPath, err := v.dm.AttachAndMount(op, diskDsURI, true)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		e1 := v.dm.UnmountAndDetach(op, diskDsURI, true)
```

Comment:
if you want to assign `err` here in this fashion you will still need to check that error after the defer. Might be better to log the Error here instead of reassigning it. It is entirely possible that this err will be related to a refcount err in which case logging the error would be very helpful to us.